toolkit/tools/pkg/imagecustomizerlib/customizebootloader.go

```diff
@@ -16,21 +16,38 @@ import (
 	"go.opentelemetry.io/otel"
 )
 
+var (
+	// Bootloader-related errors
+	ErrBootloaderHardReset             = NewImageCustomizerError("Bootloader:HardReset", "failed to perform hard reset")
+	ErrBootloaderKernelCommandLineAdd  = NewImageCustomizerError("Bootloader:KernelCommandLineAdd", "failed to add kernel command line")
+	ErrBootloaderSelinuxModeGet        = NewImageCustomizerError("Bootloader:SelinuxModeGet", "failed to get SELinux mode")
+	ErrBootloaderRootFilesystemFind    = NewImageCustomizerError("Bootloader:RootFilesystemFind", "failed to find root filesystem")
+	ErrBootloaderRootMountIdTypeGet    = NewImageCustomizerError("Bootloader:RootMountIdTypeGet", "failed to get root mount ID type")
+	ErrBootloaderImageBootTypeGet      = NewImageCustomizerError("Bootloader:ImageBootTypeGet", "failed to get image boot type")
+	ErrBootloaderDiskConfigure         = NewImageCustomizerError("Bootloader:DiskConfigure", "failed to configure disk")
+	ErrBootloaderRootMountFind         = NewImageCustomizerError("Bootloader:RootMountFind", "failed to find root mount")
+	ErrBootloaderRootMountSourceParse  = NewImageCustomizerError("Bootloader:RootMountSourceParse", "failed to parse root mount source")
+	ErrBootloaderVerityRootUnsupported = NewImageCustomizerError("Bootloader:VerityRootUnsupported", "verity root unsupported")
+	ErrBootloaderMountIdUnsupported    = NewImageCustomizerError("Bootloader:MountIdUnsupported", "mount ID type unsupported")
+)
+
 func handleBootLoader(ctx context.Context, baseConfigPath string, config *imagecustomizerapi.Config, imageConnection *imageconnection.ImageConnection,
 	partUuidToFstabEntry map[string]diskutils.FstabEntry, newImage bool,
 ) error {
 	switch {
 	case config.OS.BootLoader.ResetType == imagecustomizerapi.ResetBootLoaderTypeHard || newImage:
 		err := hardResetBootLoader(ctx, baseConfigPath, config, imageConnection, partUuidToFstabEntry, newImage)
 		if err != nil {
-			return fmt.Errorf("failed to hard reset bootloader:\n%w", err)
+			return fmt.Errorf("%w: %w", ErrBootloaderHardReset,
+				fmt.Errorf("failed to hard reset bootloader:\n%w", err))
 		}
 
 	default:
 		// Append the kernel command-line args to the existing grub config.
 		err := AddKernelCommandLine(ctx, config.OS.KernelCommandLine.ExtraCommandLine, imageConnection.Chroot())
 		if err != nil {
-			return fmt.Errorf("failed to add extra kernel command line:\n%w", err)
+			return fmt.Errorf("%w: %w", ErrBootloaderKernelCommandLineAdd,
```

Comment:
the errors like these in this file should just use the error objects defined, we don't need the extra fmt.Errorf nested inside